src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -92,157 +75,184 @@ internal sealed class ParsedStackFrame : ParsedFrame
+                foreach (var document in project.Documents)
+                {
+                    if (document.FilePath == fileName)
+                    {
+                        return ImmutableArray.Create(document);
+                    }
+
+                    else if (document.Name == documentName)
+                    {
+                        potentialMatches.Add(document);
+                    }
+                }
             }
 
-            return OriginalText[(lastSpan.End + 1)..];
+            return potentialMatches.ToImmutableArray();
         }
 
-        /// <summary>
-        /// If the frame has file information, gets the text between the method and file information.
-        /// ex: at ConsoleApp4.MyClass.M[T](T t) in [|C:\repos\Test\MyClass.cs:line 7|]
-        /// </summary>
-        public string? GetFileText()
+        private static bool MatchParameters(ImmutableArray<IParameterSymbol> parameters, StackFrameParameterList stackFrameParameters)
         {
-            if (FileSpan == default)
+            if (parameters.Length != stackFrameParameters.Parameters.Length)
             {
-                return null;
+                return false;
             }
 
-            return OriginalText[FileSpan.Start..FileSpan.End];
-        }
-
-        /// <summary>
-        /// If the frame has file information, gets the text between the method and file information.
-        /// ex: at ConsoleApp4.MyClass.M[T](T t)[| in |]C:\repos\Test\MyClass.cs:line 7
-        /// </summary>
-        public string? GetTextBetweenTypeAndFile()
-        {
-            if (FileSpan == default)
+            for (var i = 0; i < stackFrameParameters.Parameters.Length; i++)
             {
-                return null;
+                var stackFrameParameter = stackFrameParameters.Parameters[i];
+                var paramSymbol = parameters[i];
+
+                if (paramSymbol.Name != stackFrameParameter.Identifier.ToString(skipTrivia: true))
+                {
+                    return false;
+                }
+
+                if (!MatchType(paramSymbol.Type, stackFrameParameter.Type))
+                {
+                    return false;
+                }
             }
 
-            return OriginalText[(ArgsSpan.End + 1)..FileSpan.Start];
+            return true;
         }
 
-        internal (Document? document, int line) GetDocumentAndLine(Solution solution)
+        private static bool MatchTypeArguments(ImmutableArray<ITypeSymbol> typeArguments, StackFrameTypeArgumentList? stackFrameTypeArgumentList)
         {
-            var fileMatches = GetFileMatches(solution, out var lineNumber);
-            if (fileMatches.IsEmpty)
+            if (stackFrameTypeArgumentList is null)
             {
-                return (null, 0);
+                return typeArguments.IsDefaultOrEmpty;
             }
 
-            return (fileMatches.First(), lineNumber);
+            if (typeArguments.IsDefaultOrEmpty)
```

Comment:
should never be Default.  can you remove that part (i prefer code to not try to be resilient to things that should never happen).